src/form-data/ReviewPage.tsx

```diff
@@ -0,0 +1,181 @@
+import React, { useContext, useEffect, useLayoutEffect } from 'react';
+import { FormikContextType, useFormikContext } from 'formik';
+import { Link } from 'react-router-dom';
+import { VaOnThisPage } from '@department-of-veterans-affairs/component-library/dist/react-bindings';
+import { Page, PageProps } from '../routing';
+import { FieldObject, PageObject } from './types';
+
+import { parseDate } from '../utils/helpers';
+import { PageContext } from './PageContext';
+import { type } from '@testing-library/user-event/dist/type';
+
+const TruthAndFalseLabels = [
+  {
+    label: 'Yes',
+    values: [true, 1],
+  },
+  {
+    label: 'No',
+    values: [false, 0],
+  },
+];
+
+/**
+ * Transforms fields value into value that is more readable
+ * @param {boolean | object | string} field object value
+ * @param {string} key object key name
+ *
+ * @beta
+ */
+const transformFieldValue = (key: number, field: FieldObject) => {
+  if (
+    typeof field?.value === 'undefined' ||
+    field.value === null ||
+    typeof field?.value === 'object'
+  )
+    return;
+
+  // check for a 0,false,1,or true value
+  const truthOrFalseIndex = TruthAndFalseLabels.findIndex((labelValue) => {
+    return !!labelValue.values.find(
+      (fieldValue) => field?.value === fieldValue
+    );
+  });
+  if (truthOrFalseIndex >= 0) {
+    return TruthAndFalseLabels[truthOrFalseIndex].label;
+  }
+
+  // Add field options
+  if (field?.options) {
+    const fieldIndex = field.options.findIndex(
+      (fieldOption) => fieldOption.value === field?.value
+    );
+    if (fieldIndex >= 0) {
+      return field.options[fieldIndex].label;
+    }
+  }
+  // Parse date
+  if (field?.type === 'date') {
+    const date = parseDate(field.value as string);
+    return date.toLocaleDateString('en-US', {
+      day: 'numeric',
+      month: 'long',
+      year: 'numeric',
+    });
+  }
+  // Number fields
+  if (field?.type === 'accounting') {
+    return `$${field.value as string}`;
+  }
+  // Telephone fields
+  if (field?.type === 'tel') {
+    return `${(field.value as string).replace(
+      /(\d{3})(\d{3})(\d{4})/,
+      '($1) $2-$3'
+    )}`;
+  }
+  return field.value;
+};
+
+/**
+ * Loops through field objects and adds JSX to a buffer
+ * to be rendered later
+ * @param {objects} fields object value
+ * @param {number} rank level number of recursion
+ *
+ * @beta
+ */
+const bufferFields = (fields: FieldObject[], rank = 0) => {
+  const fieldBuffer: JSX.Element[] = [];
+  fields.forEach((field, key) => {
+    const fieldJSX = recurseField(key, field, rank);
```

Comment:
This problem came up in vets-website because of the linting rules, but is there a way we can use recursion without calling upon `recurseField` which isn't defined until after this function? 

I believe I had to put in a linter disable comment to get this to work inside of vet-website.